packages/forms/frontend/sirius-components-forms/src/representations/useFormSubscription.tsx

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+
+import { gql, OnDataOptions, useSubscription } from '@apollo/client';
+import { useMultiToast } from '@eclipse-sirius/sirius-components-core';
+import { useEffect, useState } from 'react';
+import { formRefreshedEventPayloadFragment } from '../form/FormEventFragments';
+import {
+  GQLFormEventInput,
+  GQLFormEventSubscription,
+  GQLFormEventVariables,
+  UseFormSubscriptionState,
+  UseFormSubscriptionValue,
+} from './useFormSubscription.types';
+
+export const formEventSubscription = `
+  subscription formEvent($input: FormEventInput!) {
+    formEvent(input: $input) {
+      __typename
+      ... on FormRefreshedEventPayload {
+        ...formRefreshedEventPayloadFragment
+      }
+    }
+  }
+  ${formRefreshedEventPayloadFragment}
+`;
+
+export const useFormSubscription = (editingContextId: string, formId: string): UseFormSubscriptionValue => {
+  const [state, setState] = useState<UseFormSubscriptionState>({
+    id: crypto.randomUUID(),
+    complete: false,
+    payload: null,
+  });
+
+  const input: GQLFormEventInput = {
+    id: state.id,
+    editingContextId,
+    formId,
+  };
+
+  const variables: GQLFormEventVariables = { input };
+
+  const onData = ({ data }: OnDataOptions<GQLFormEventSubscription>) => {
+    const { data: gqlFormEventSubscription } = data;
+    if (gqlFormEventSubscription) {
+      const { formEvent: payload } = gqlFormEventSubscription;
+      setState((prevState) => ({ ...prevState, payload }));
+    }
+  };
+
+  const onComplete = () => setState((prevState) => ({ ...prevState, complete: true }));
+
+  const { error, loading } = useSubscription<GQLFormEventSubscription, GQLFormEventVariables>(
```

Comment:
Why not destructure data here like that `{ data, error, loading }` and remove the payload from the state? You would also remove one re-rendering with the setState line57 which would disappear.

Reply:
Okay I modified all the custom hook to act like this.